### PR TITLE
Sms gateway registration fix

### PR DESCRIPTION
**Pull request by alanhartless**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

https://github.com/mautic/mautic/pull/5983 caused the Twilio to not show up in the SMS gateway dropdown. It happened to work in 2.13.0 because 

```
'tags'  => [
                    'name' => 'mautic.sms_transport', 'Twilio',
                ],

```
 
resulted in 

```
/app/bundles/CoreBundle/DependencyInjection/Compiler/SmsTransportPass.php:44:
array (size=2)
  0 => string 'mautic.sms_transport' (length=20)
  1 => string 'Twilio' (length=6)
```

so that https://github.com/mautic/mautic/blob/8b993d8eae8ff8b59f3bab1777369dd25938824e/app/bundles/SmsBundle/Sms/TransportChain.php#L127 ignored `mautic.sms_transport` but found `Twilio`. 

But by making it a single tag, `Twilio` was never used to find the integration to check if it is published. The change would require the integration name itself to be `mautic.sms_transport` which isn't possible due to multiple SMS integrations needing to be registered as transports.
 
This PR fixes it so that it's possible to define the integration alias through tagArguments (attributes) or default to the service ID if none is defined. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. In staging or release-2.13.1 branch, go to plugins -> click on Twilio, enable it and enter some fake details. Go to Configuration -> Text Message Settings -> and notice Twilio is not listed in the default transport drop down. 

#### Steps to test this PR:
1. Repeat and Twilio will be in the default transport dropdown

### CODE REVIEW

**Comment by javjim-mautic** [comment]
Was able to reproduce bug, and fix, great work

File: app/bundles/SmsBundle/Config/config.php [hunk 1]
```diff
@@ -100,19 +100,25 @@
                 'alias' => 'sms_api',
             ],
             'mautic.sms.transport_chain' => [
-                'class'     => '\Mautic\SmsBundle\Sms\TransportChain',
-                'arguments' => ['%mautic.sms_transport%', 'mautic.helper.integration', 'monolog.logger.mautic'],
+                'class'     => \Mautic\SmsBundle\Sms\TransportChain::class,
```
Comment (galvani) [inline app/bundles/SmsBundle/Config/config.php]: why is this? i usually don't do it, i like to keep configuration static. i believe it should be text.

Comment (Maxell92) [reply app/bundles/SmsBundle/Config/config.php]: We this this format almost everywhere. Only the old code is written with strings - `::class` is used when Mautic dropped support of old PHP.

**Comment by Maxell92** [comment]
Tested, reproduced bug and bugfix works for me.

**Pull request merged**